virttest/qemu_monitor.py

```diff
@@ -113,7 +113,8 @@ def get_monitor_filename(vm, monitor_name):
     :param monitor_name: The monitor name.
     :return: The string of socket file name for qemu monitor.
     """
-    return os.path.join(data_dir.get_tmp_dir(),
+    basedir = vm.params.get("persistent_tmp_path", data_dir.get_tmp_dir())
```

Comment:
it should not affect the storage location of test log files.

Reply:
Hi @xutian, can you elaborate on what you mean? If the "persistent_tmp_path" parameter is not set, the behavior should be identical to the one before. In order to be able to keep vms for longer than a test run we have to preserve the monitor filename together with others and thus have to change to a permanent location.

Reply:
@pevogam 
If the persistent_tmp_path parameter is configured, the serial and monitor's logs are saved in the defined directory, but other test log files (for example, debug files) are still saved in the default location. This is not a good design.

Reply:
If by other files you mean the job log, test results and and so on, then I tend to disagree. The point here is *precisely* to keep only the files related to the persistence and communication with the vm object permanent (env file, monitor socket and so on) beyond tests and all these other files are test/job specific.

We do not store permanently any log files related to the serial console and monitor at least according to my checks but only the socket files which we need beyond a single test/job.

Reply:
@pevogam 
If a VM is reused many times in a round of testing, but in one of the tests there is an abnormal state and we need to analyze the serial port log, but since all serial port logs are in the same file, this will give Analysis of the logs creates difficulties. This is my concern

Reply:
Hi @xutian can you provide me with pointers for the log filenames you are referring to? This is the file structure after a job run with the patch above applied:

```
# ls -1 <persistent_tmp_path>/
address_pool
address_pool.lock
env
monitor-catch_monitor-20190704-112820-jZnnwm3a
monitor-catch_monitor-20190708-051444-wD5AWvfT
monitor-catch_monitor-20190713-021459-puWT6b9d
monitor-catch_monitor-20190713-195338-GSytHpRz
monitor-catch_monitor-20190713-235755-MEg3TO1E
monitor-catch_monitor-20190714-143311-ncY8HKXJ
monitor-hmp1-20190704-112820-jZnnwm3a
monitor-hmp1-20190708-051444-wD5AWvfT
monitor-hmp1-20190713-021459-puWT6b9d
monitor-hmp1-20190713-195338-GSytHpRz
monitor-hmp1-20190713-235755-MEg3TO1E
monitor-hmp1-20190714-143311-ncY8HKXJ
seabios-20190704-112820-jZnnwm3a
seabios-20190708-051444-wD5AWvfT
seabios-20190713-021459-puWT6b9d
seabios-20190713-195338-GSytHpRz
seabios-20190713-235755-MEg3TO1E
seabios-20190714-143311-ncY8HKXJ
serial-serial1-20190704-112820-jZnnwm3a
serial-serial1-20190708-051444-wD5AWvfT
serial-serial1-20190713-021459-puWT6b9d
serial-serial1-20190713-195338-GSytHpRz
serial-serial1-20190713-235755-MEg3TO1E
serial-serial1-20190714-143311-ncY8HKXJ

ls -1 <results>/latest/test-results/<test-variant>/
catch_monitor-vm1.log
catch_monitor-vm2.log
catch_monitor-vm3.log
catch_monitor-vm4.log
data
debug.log
hmp1-vm1.log
hmp1-vm2.log
hmp1-vm3.log
hmp1-vm4.log
ip-sniffer.log
output
remote.log
screendumps_vm1_3206_iter0
screendumps_vm2_9319_iter0
screendumps_vm3_38756_iter0
screendumps_vm4_36179_iter0
stderr
stdout
sysinfo
vm_register_vm1_3206
vm_register_vm2_9319
vm_register_vm3_38756
vm_register_vm4_36179
whiteboard
```

From what I gather you are referring to the monitor-related logs like `catch_monitor` and `hmp1`? If so they are already within the test-specific result folder as you suggested. Regarding the serial log file scenario you described, the same thing should apply for it too:

```python
        log_name = "serial-%s-%s.log" % (tmp_serial, self.name)
        self.serial_console_log = os.path.join(utils_misc.get_log_file_dir(),
                                               log_name)
        self.serial_console = aexpect.ShellSession(
            "nc -U %s" % self.get_serial_console_filename(tmp_serial),
```

In this snippet you can see that the serial log file is set from `utils_misc` and only the socket is modified by the optional permanent path in the `get_serial_console_filename` method call.

Reply:
Hi @xutian, is the above answer satisfactory for your inquiries?

Reply:
Looks ok